src/Util/ComposerAutoloaderFinder.php

```diff
@@ -19,45 +19,87 @@
  */
 class ComposerAutoloaderFinder
 {
+    private $rootNamespace;
+
     /**
      * @var ClassLoader|null
      */
     private $classLoader = null;
 
+    public function __construct(string $rootNamespace)
+    {
+        $this->rootNamespace = [
+            'psr0' => rtrim($rootNamespace, '\\'),
+            'psr4' => rtrim($rootNamespace, '\\').'\\',
+        ];
+    }
+
     public function getClassLoader(): ClassLoader
     {
         if (null === $this->classLoader) {
-            $this->classLoader = $this->findComposerClassLoader();
+            $this->findComposerClassLoader();
         }
 
         if (null === $this->classLoader) {
-            throw new \Exception('Composer ClassLoader not found!');
+            throw new \Exception("Could not find a Composer autoloader that autoloads from '{$this->rootNamespace['psr4']}'");
         }
 
         return $this->classLoader;
     }
 
-    /**
-     * @return ClassLoader|null
-     */
     private function findComposerClassLoader()
     {
         $autoloadFunctions = spl_autoload_functions();
 
         foreach ($autoloadFunctions as $autoloader) {
-            if (\is_array($autoloader) && isset($autoloader[0]) && \is_object($autoloader[0])) {
-                if ($autoloader[0] instanceof ClassLoader) {
-                    return $autoloader[0];
-                }
-
-                if ($autoloader[0] instanceof DebugClassLoader
-                    && \is_array($autoloader[0]->getClassLoader())
-                    && $autoloader[0]->getClassLoader()[0] instanceof ClassLoader) {
-                    return $autoloader[0]->getClassLoader()[0];
-                }
+            $classLoader = $this->extractComposerClassLoader($autoloader);
+            if ($classLoader && $this->locateMatchingClassLoader($classLoader)) {
+                return;
+            }
```

Comment:
Here, if we make `locateMatchingClassLoader()` return a value, we can simplify a bit (at least, simpler for how I think!)

```php
if (null === classLoader) {
    continue;
}

$finalClassLoader = $this->locateMatchingClassLoader($classLoader);

if (null !== $finalClassLoader) {
    return $finalClassLoader;
}
```